use-cases/vehicle-routing/src/main/resources/META-INF/resources/app.js

```diff
@@ -336,6 +337,116 @@ function renderTimelines(routePlan) {
+            const analysisTHead = $(`<thead/>`).append($(`<tr/>`)
+                .append($(`<th></th>`))
+                .append($(`<th>Constraint</th>`).css({textAlign: 'left'}))
+                .append($(`<th>Type</th>`))
+                .append($(`<th># Matches</th>`))
+                .append($(`<th>Weight</th>`))
+                .append($(`<th>Score</th>`))
+                .append($(`<th></th>`)));
+            analysisTable.append(analysisTHead);
+            const analysisTBody = $(`<tbody/>`)
+            $.each(scoreAnalysis.constraints, (index, constraintAnalysis) => {
+                let icon = constraintAnalysis.type == "hard" && constraintAnalysis.implicitScore < 0 ? '<span class="fas fa-exclamation-triangle" style="color: red"></span>' : '';
+                if (!icon) icon = constraintAnalysis.weight < 0 && constraintAnalysis.matches.length == 0 ? '<span class="fas fa-check-circle" style="color: green"></span>' : '';
+
+                let row = $(`<tr/>`);
+                row.append($(`<td/>`).html(icon))
+                    .append($(`<td/>`).text(constraintAnalysis.name).css({textAlign: 'left'}))
+                    .append($(`<td/>`).text(constraintAnalysis.type))
+                    .append($(`<td/>`).html(`<b>${constraintAnalysis.matches.length}</b>`))
+                    .append($(`<td/>`).text(constraintAnalysis.weight))
+                    .append($(`<td/>`).text(constraintAnalysis.implicitScore));
+
+                analysisTBody.append(row);
+
+                if (constraintAnalysis.matches.length > 0) {
+                    let matchesRow = $(`<tr/>`).addClass("collapse").attr("id", "row" + index + "Collapse");
+                    let matchesListGroup = $(`<ul/>`).addClass('list-group').addClass('list-group-flush').css({textAlign: 'left'});
+
+                    $.each(constraintAnalysis.matches, (index2, match) => {
+                        matchesListGroup.append($(`<li/>`).addClass('list-group-item').addClass('list-group-item-light').text(match.justification.description));
+                    });
+
+                    matchesRow.append($(`<td/>`));
+                    matchesRow.append($(`<td/>`).attr('colspan', '6').append(matchesListGroup));
+                    analysisTBody.append(matchesRow);
+
+                    row.append($(`<td/>`).append($(`<a/>`).attr("data-toggle", "collapse").attr('href', "#row" + index + "Collapse").append($(`<span/>`).addClass('fas').addClass('fa-arrow-down')).click(e => {
+                        matchesRow.collapse('toggle');
+                        let target = $(e.target);
+                        if (target.hasClass('fa-arrow-down')) {
+                            target.removeClass('fa-arrow-down').addClass('fa-arrow-up');
+                        } else {
+                            target.removeClass('fa-arrow-up').addClass('fa-arrow-down');
+                        }
+                    })));
```

Comment:
This might possibly go into a separate function.

Reply:
Refactored.